File: Week01/1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		real_target := target - v
		ele, exist := m[real_target]
		m[v] = i
		if exist {
			return []int{ele, i}
		}
	}
	return []int{}
}
// @lc code=end

File: Week02/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for idx, num in enumerate(nums):
            res = target - num
            if res in d:
                return [d[res], idx]
            d[num] = idx
        return [0, 0]
# @lc code=end

